### DISCUSSION TITLE
Adding a new hook between parse and func for commands

### DISCUSSION THREAD

**Discussion by pruuk** [evennia/evennia#2630] [Evennia Questions] [answered]
If I wanted to add a new hook between parse and func for commands, how would I go about doing it?

Sometimes I want to run checks after parsing (based upon the parsing, but before I run a command's function. I'd rather not re-write the parser in  the at_pre_cmd hook in order to do so.

**Comment by Tegiminis** [comment, marked as answer]
Two ways. In both, you need to add a hook method to your command object. (let's call it `after_parse`)

1. Call the hook method at the end of the parse hook. This means it'll always run at the end of a command parse. You'll have to do this for each command you want to have this behavior.
2. Call the hook method in the `_run_command` method in the CmdHandler, between `yield cmd.parse` and `ret = cmd.func()`. This is the more robust solution, but necessarily involves modifying the handler.

**Reply by pruuk** [reply]
Thanks for the pointer. I appreciate it!
